DirectProgramming/C++SYCL_FPGA/Tutorials/Features/IPAuthoring_interfaces_overview/host-pipes/Readme.md

```diff
@@ -0,0 +1,305 @@
+# Host Pipe Streaming
+This example demonstrate a vector addition design using streaming date interface.
+
+This design will show how to use host pipe to stream data in and out of kernel and how to customize their properties.
+
+## Invocation Interface
+By default, an un-decorated oneAPI kernel will have all its control signals and arguments passed through an MM-Agent interface:
+
+## Data Interface - Host Pipe
+Pipes are a first-in first-out (FIFO) buffer construct that provide links between elements of a design. Pipes that connect a host and a device are referred to as host pipes. In this design, interfaces `A_in`, `B_in`, `C_in` are implemented as Host Pipes.
+
+Host pipe support is enabled by including the following include statement in your design:
+```cpp
+#include <sycl/ext/intel/experimental/pipes.hpp>
+```
+### Declear Host Pipe
+Each individual host pipe is a program scope class declaration of the templated pipe class:
+```cpp
+sycl::ext::intel::experimental::pipe<id, type, min_capacity, properties>;
+```
+
+The table shows how to configure each templated parameter:
```

Comment:
these details should be covered in a specific tutorial or documentation. you don't need all this here.